CenterForOpenScience/osf.io
# [PLAT-1025] Pass X-CSRFToken header from legacy frontend

Paired with @aaxelb 

**Purpose**

Fix requests to APIv2 from legacy frontend when
CSRF validation is enforced.

**Changes**

* Modify osfHelpers.ajaxJSON and mithrilHelpers.apiV2Config
  to transparently add the X-CSRFToken header for non-GET
  requests to APIv2
* Modify usages of m.request and $.ajax to use the above helpers
* Add Flask handlers to set the API CSRF token on every
  session-authenticated request
* Remove some unused JS I found while searching through
  the codebase

**QA Notes**

This will affect all legacy frontend forms that send
a request to APIv2. This includes (not exhaustive):

* Creating component
* All actions on the "My Projects" page
* Accept Terms of Service alert at top of page
* Bulk node deletion on Project Settings
* Editing title/description on Project Settings
* Making projects/components public/private on Project Overview
* Check in/Check out on Files page
* Check in/Check out on File Detail page

NOTE: There may be more--these are only the ones I ran into
while writing this patch.

**Ticket**

https://openscience.atlassian.net/browse/PLAT-1025
